MapleServer2/PacketHandlers/Game/Helpers/ItemUseHelper.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using Maple2Storage.Types.Metadata;
+using MapleServer2.Data.Static;
+using MapleServer2.Servers.Game;
+using MapleServer2.Types;
+
+namespace MapleServer2.PacketHandlers.Game.Helpers
+{
+    public static class ItemUseHelper
+    {
+        private enum BoxType : byte
+        {
+            OPEN = 0x00,
+            SELECT = 0x01
+        }
+
+        public static void OpenBox(GameSession session, List<ItemContent> content)
+        {
+            Random rng = new Random();
+            bool oneGroup = true;
+
+            foreach (ItemContent item in content)
+            {
+                if (item.DropGroup != content[0].DropGroup)
+                {
+                    oneGroup = false;
+                    break;
+                }
+            }
+
+            if (oneGroup)
+            {
+                int smartDropRate = content[0].SmartDropRate;
+
+                if (smartDropRate == 0)
+                {
+                    int rand = rng.Next(0, content.Count);
+                    GiveItem(session, content[rand]);
+                }
+                else if (smartDropRate == 100)
+                {
+                    foreach (ItemContent item in content)
+                    {
+                        if (ItemMetadataStorage.GetRecommendJobs(item.Id).Contains(session.Player.JobGroupId) || ItemMetadataStorage.GetRecommendJobs(item.Id).Contains(0))
```

Comment:
We call `ItemMetadataStorage.GetRecommendJobs(item.Id)` around 6 times in this method, perhaps we should set this to a variable to shorten calls. Perhaps also include a comment on what it means when it contains `0`